lib/util/data_view_writer.js

```diff
@@ -0,0 +1,171 @@
+ * @struct
+ * @constructor
+ * @export
+ */
+shaka.util.DataViewWriter = function(dataView, endianness) {
+  /** @private {!DataView} */
+  this.dataView_ = dataView;
+
+  /** @private {!Uint8Array} */
+  this.byteArray_ = new Uint8Array(dataView.buffer);
+
+  /** @private {boolean} */
+  this.littleEndian_ =
+      endianness == shaka.util.DataViewWriter.Endianness.LITTLE_ENDIAN;
+
+  /** @private {number} */
+  this.position_ = 0;
+};
+
+
+/**
+ * Endianness.
+ * @enum {number}
+ * @export
+ */
+shaka.util.DataViewWriter.Endianness = {
+  BIG_ENDIAN: 0,
+  LITTLE_ENDIAN: 1
+};
+
+
+/**
+ * @return {boolean} True if the writer has more data, false otherwise.
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.hasMoreData = function() {
+  return this.position_ < this.dataView_.byteLength;
+};
+
+
+/**
+ * Gets the current byte position.
+ * @return {number}
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.getPosition = function() {
+  return this.position_;
+};
+
+
+/**
+ * Gets the byte length of the DataView.
+ * @return {number}
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.getLength = function() {
+  return this.dataView_.byteLength;
+};
+
+/**
+ * Gets the byte length of the DataView.
+ * @return {!Uint8Array}
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.getArray = function() {
+  return this.byteArray_;
+};
+
+
+/**
+ * Writes an unsigned 8 bit integer, and advances the writer.
+ * @param {number} value The integer.
+ * @throws {shaka.util.Error} when writing past the end of the data view.
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.writeUint8 = function(value) {
+  try {
+    this.dataView_.setUint8(this.position_, value);
+  } catch (exception) {
+    this.throwOutOfBounds_();
+  }
+  this.position_ += 1;
+};
+
+
+/**
+ * Writes an unsigned 16 bit integer, and advances the writer.
+ * @param {number} value The integer.
+ * @throws {shaka.util.Error} when writing past the end of the data view.
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.writeUint16 = function(value) {
+  try {
+    this.dataView_.setUint16(this.position_, value, this.littleEndian_);
+  } catch (exception) {
+    this.throwOutOfBounds_();
+  }
+  this.position_ += 2;
+};
+
+
+/**
+ * Writes an unsigned 32 bit integer, and advances the writer.
+ * @param {number} value The integer.
+ * @throws {shaka.util.Error} when writing past the end of the data view.
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.writeUint32 = function(value) {
+  try {
+    this.dataView_.setUint32(this.position_, value, this.littleEndian_);
+  } catch (exception) {
+    this.throwOutOfBounds_();
+  }
+  this.position_ += 4;
+};
+
+/**
+ * Writes an unsigned 32 bit integer, and advances the writer.
+ * @param {Uint8Array} array The integer.
+ * @throws {shaka.util.Error} when writing past the end of the data view.
+ * @export
+ */
+shaka.util.DataViewWriter.prototype.put = function(array) {
+  try {
+    this.byteArray_.set(array, this.position_);
```

Comment:
Closure compiler throws an error here. Uint8Array.set can get type array as well as you can see in the [doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/set) . Alternatively, I can loop it and set it as one byte. However, this seems more reasonable for me. Any thoughts?